.github/renovate.json

```diff
@@ -0,0 +1,208 @@
+{
+  "extends": [
+    "config:base"
+  ],
+  "enabledManagers": [
+    "npm"
+  ],
+  "force": {
+    "constraints": {
+      "node": "< 15.0.0",
+      "npm": "< 7.0.0"
+    }
+  },
+  "pruneStaleBranches": false,
+  "rangeStrategy": "bump",
+  "commitMessagePrefix": "patch:",
+  "commitBody": "Update {{depName}} from {{currentVersion}} to {{newVersion}}\n\nChange-type: patch",
+  "prHourlyLimit": 0,
+  "labels": [
+    "dependencies"
+  ],
+  "ignoreDeps": [
+    "node"
+  ],
+  "packageRules": [
+    {
+      "groupName": "storybook-non-major",
+      "matchPackagePrefixes": ["@storybook"],
+      "commitBody": "Change-type: patch",
+      "matchUpdateTypes": [
+        "minor",
+        "patch"
+      ],
+      "automerge": true
+    },
+    {
+      "groupName": "storybook-major",
+      "matchPackagePrefixes": ["@storybook"],
+      "commitBody": "Change-type: patch",
+      "matchUpdateTypes": [
+        "major"
+      ],
+      "automerge": false
+    },
+    {
+      "groupName": "types",
+      "matchPackagePrefixes": ["@types"],
+      "commitBody": "Change-type: patch",
+      "matchUpdateTypes": [
+        "major",
+        "minor",
+        "patch"
```

Comment:
I believe you should be able to remove `matchUpdateTypes` from rule groups where you are including all update types. By doing so, the rule should apply to all possible update types whose package prefix matches `@types` for example.

Reply:
Hey @joshbwlng, that was my first thought as well but then, checking the documentation I've seen that matchUpdateTypes can include `major,minor,patch,pin,digest,lockFileMaintenance,rollback,bump`. Not sure if we want all of them, isn't it ?

Reply:
@JSReds I don't think you're wrong, but I believe it ultimately depends on `rangeStrategy` and the way package versions are specified in `package.json`. For example, in Jellyfish repos we use `^x.y.z` for versions and `bump` for `rangeStrategy` so we don't receive `pin` type updates from Renovate. I've only ever seen `patch`, `minor`, and `major` updates from Renovate with our setup. That being said, there's no harm in keeping the configs this way to play it safe. If you for some reason do receive `pin`, `digest` or other updates, they would still be opened, but in separate PRs as they don't fall into any of your package rules.

Reply:
Got it, makes perfect sense @joshbwlng, thank you for the explanation